sequence_processing_pipeline/ConvertJob.py

```diff
@@ -210,3 +222,152 @@ def parse_job_script(job_script_path):
+                      r"_S\d+_L\d+_[R,I]\d+_\d+.fastq.gz$"
+                for full_path in fastq_paths:
+                    file_path, file_name = split(full_path)
+                    if re.match(rgx, file_name):
+                        curr_sample_info[MATCHING_FILES_KEY].append(full_path)
+                    # endif path matches sample
+                # next fastq path
+            # next sample
+        # next project
+
+        self.contains_replicates = sheet_needs_demuxing(sheet)
+        self.info = projects_info
+        self.controls_info = controls_info
+
+    def copy_sequences(self, sample_name, source_project, dest_project):
+        """
+        Copies all fastq files related to a sample into another project.
+        :param sample_name: A sample_name value if copy_all_replicates is
+            False; an orig_name value otherwise.
+        :param source_project: The source project name including qiita_id.
+        :param dest_project: The destination project name including qiita_id.
+        :return: None
+        """
+        if self.info is None:
+            raise ValueError("This method cannot be called until processing "
+                             "has completed.")
+
+        project_names = list(self.info.keys())
+
+        # confirm source and destination projects are both valid.
+        for proj in [source_project, dest_project]:
+            if proj not in project_names:
+                raise ValueError(f"'{proj}' is not defined in sample sheet")
+
+        if source_project == dest_project:
+            raise ValueError(f"source '{source_project}' and destination "
+                             f"'{dest_project}' projects are the same")
+
+        # note that the user can supply a sample-name that didn't make it
+        # through the conversion step and may have no files matched to it.
+        # this is considered okay and a possible outcome of conversion. In
+        # this case zero files are copied and no information is passed back
+        # to the user.
+
+        # projects that contain replicates must also be considered. In this
+        # situation, the sample_name needs to be compared against all values
+        # in the orig_name column and the individual sample_names (w/well-ids)
+        # must be discovered; those individual sample_names can then be
+        # processed.
+
+        samples = self.info[source_project][SAMPLES_DETAILS_KEY]
+
+        # Either we will need to copy all replicates, in which case we need to
+        # match our input to orig_name values, or we want to copy
+        # a single sample, in which case the input needs to be matched to
+        # sample_name values. The results of our match will be
+        # stored in results and if we are not copying all replicates, there
+        # should only be one sample matched in results.
+        results = []
+
+        if self.contains_replicates is True:
```

Comment:
We used to have a coding practices document for the lab a while ago, right? Maybe we should revisit that because I believe we're not consistent on this practice across the board. Personally I don't mind and can go either way or both.